instrumentation/cassandra/cassandra-4.4/library/src/main/java/io/opentelemetry/instrumentation/cassandra/v4_4/CassandraNetworkAttributesGetter.java

```diff
@@ -28,7 +30,12 @@ public InetSocketAddress getNetworkPeerInetSocketAddress(
     }
     // resolve() returns an existing InetSocketAddress, it does not do a dns resolve,
     // at least in the only current EndPoint implementation (DefaultEndPoint)
-    SocketAddress address = coordinator.getEndPoint().resolve();
-    return address instanceof InetSocketAddress ? (InetSocketAddress) address : null;
+    EndPoint endPoint = coordinator.getEndPoint();
+    if (endPoint instanceof DefaultEndPoint) {
+      return (InetSocketAddress) coordinator.getEndPoint().resolve();
+    } else if (endPoint instanceof SniEndPoint) {
+      return ((SniEndPoint) endPoint).resolve();
```

Comment:
can you fix up the comment, and also apply to the other locations (since reverse dns lookup is a tricky hidden side effect that's not obvious from the code)